versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java

```diff
@@ -1236,6 +1268,184 @@ protected RefLogEntry writeRefLogEntry(
     return refLogEntry;
   }
 
+  protected Map<String, String> compactGlobalLog() {
+    // Not using casOpLoop() here, as it is simpler than adopting casOpLoop().
+    try (TryLoopState tryState =
+        newTryLoopState(
+            "compact-global-log",
+            ts ->
+                repoDescUpdateConflictMessage(
+                    String.format(
+                        "%s after %d retries, %d ms",
+                        "Retry-Failure", ts.getRetries(), ts.getDuration(TimeUnit.MILLISECONDS))),
+            this::tryLoopStateCompletion,
+            config)) {
+
+      CompactionStats stats = new CompactionStats();
+      while (true) {
+        NonTransactionalOperationContext ctx = NON_TRANSACTIONAL_OPERATION_CONTEXT;
+
+        GlobalStatePointer pointer = fetchGlobalPointer(ctx);
+
+        // Collect the old global-log-ids, to delete those after compaction
+        List<Hash> oldLogIds = new ArrayList<>();
+        // Map with all global contents.
+        Map<String, ByteString> globalContents = new HashMap<>();
+        // Content-IDs, most recently updated contents first.
+        List<String> contentIdsByRecency = new ArrayList<>();
+
+        // Read the global log - from the most recent global-log entry to the oldest.
+        try (Stream<GlobalStateLogEntry> globalLog = globalLogFetcher(ctx, pointer)) {
```

Comment:
Not sure whether that would reduce memory pressure.

I was thinking into the other direction and prevent (actually: "abort") compaction when it's not necessary. In other words: only compact, when there was activity.
* when there are less than "num-parents-per-global-log-entry" global log entries at all (because those would be fetched as a single page anyway)
* when there's at least one compacted global-log-entry in the first page